mne/viz/_3d.py

```diff
@@ -1434,7 +1436,10 @@ def _smooth_plot(this_time, params):
     color_ave = np.mean(colors[faces], axis=1).flatten()
     curv_ave = np.mean(params['curv'][faces], axis=1).flatten()
     # matplotlib/matplotlib#11877
-    facecolors = polyc._facecolors3d
+    if LooseVersion(matplotlib_version) < '3.3.3':
+        facecolors = polyc._facecolors3d
+    else:
+        facecolors = polyc._facecolor3d
```

Comment:
Now would be a good time to abandon the private attribute all together. I suggest we do `get_facecolor()` then `set_facecolor()`. It will perhaps be slower than in-place setting, but probably not a meaningful slowdown compared to the actual matplotlib 3D rendering speed.

Can you see if it works?

Reply:
With the naive:

```py
    facecolors = polyc.get_facecolor()
    # create colors here
    facecolors[:] = colors
    polyc.set_facecolor(facecolors)
```

Locally I see:

```
mne/viz/_3d.py:1530: in _plot_mpl_stc
    _smooth_plot(initial_time, params)
mne/viz/_3d.py:1437: in _smooth_plot
    facecolors = polyc.get_facecolor()
../anaconda3/envs/mne/lib/python3.8/site-packages/mpl_toolkits/mplot3d/art3d.py:817: in get_facecolor
    return self._facecolors2d
E   AttributeError: 'Poly3DCollection' object has no attribute '_facecolors2d'
```

Reply:
`polyc.set_facecolors(colors)` without any `get_facecolor` or `get_facecolors` call seems to at least not kill the tests.